example-app/src/main/java/com/adyen/checkout/example/ui/bacs/BacsFragment.kt

```diff
@@ -0,0 +1,139 @@
+    private val binding: FragmentBacsBinding get() = requireNotNull(_binding)
+
+    private val viewModel: BacsViewModel by viewModels()
+
+    private var bacsComponent: BacsDirectDebitComponent? = null
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        // Insert return url in extras, so we can access it in the ViewModel through SavedStateHandle
+        val returnUrl = RedirectComponent.getReturnUrl(requireActivity().applicationContext)
+        arguments = bundleOf(RETURN_URL_EXTRA to returnUrl)
+
+        _binding = FragmentBacsBinding.inflate(inflater)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        viewModel.paymentMethodFlow
+            .flowWithLifecycle(viewLifecycleOwner.lifecycle)
+            .onEach { setupBacsComponent(it) }
+            .launchIn(lifecycleScope)
+
+        viewModel.viewState
+            .flowWithLifecycle(viewLifecycleOwner.lifecycle)
+            .onEach { onViewState(it) }
+            .launchIn(lifecycleScope)
+
+        viewModel.events
+            .flowWithLifecycle(viewLifecycleOwner.lifecycle)
+            .onEach { onEvent(it) }
+            .launchIn(lifecycleScope)
+    }
+
+    private fun setupBacsComponent(paymentMethod: PaymentMethod) {
+        val bacsComponent = BacsDirectDebitComponent.PROVIDER.get(
+            this,
+            paymentMethod,
+            checkoutConfigurationProvider.getBacsConfiguration(),
+            requireActivity().application,
+        )
+
+        this.bacsComponent = bacsComponent
+
+        binding.componentView.attach(bacsComponent, viewLifecycleOwner)
+
+        bacsComponent.observe(viewLifecycleOwner, viewModel::onPaymentComponentEvent)
+    }
+
+    private fun onViewState(state: BacsViewState) {
+        when (state) {
+            is BacsViewState.Error -> {
+                binding.errorView.isVisible = true
+                binding.errorView.setText(state.message)
+                binding.componentView.isVisible = false
+                binding.progressIndicator.isVisible = false
+            }
+            BacsViewState.Loading -> {
+                binding.errorView.isVisible = false
+                binding.componentView.isVisible = false
+                binding.progressIndicator.isVisible = true
+            }
+            BacsViewState.ShowComponent -> {
+                binding.errorView.isVisible = false
+                binding.componentView.isVisible = true
+                binding.progressIndicator.isVisible = false
+            }
+        }
+    }
+
+    private fun onEvent(event: BacsEvent) {
+        when (event) {
+            is BacsEvent.AdditionalAction -> bacsComponent?.handleAction(event.action, requireActivity())
+            BacsEvent.Invalid -> binding.componentView.highlightValidationErrors()
```

Comment:
Just double checking, if `BacsEvent.Invalid` was emitted multiple times in succession, all of them will be handled here because the flow is not a state flow, is that correct?

Reply:
Yes that's right